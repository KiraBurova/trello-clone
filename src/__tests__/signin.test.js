import React from 'react';

import { render, fireEvent, act } from '@testing-library/react';

import userEvent from '@testing-library/user-event';

import SignInScreen from '../screens/SignIn';

const mockFirebase = {
  auth: jest.fn(() => mockFirebase),
  signInWithEmailAndPassword: jest.fn(() => Promise.resolve('fakeResponse')),
};

jest.mock('react-router-dom', () => {
  return {
    Redirect: jest.fn(({ to }) => `Redirected to ${to}`),
    useHistory: () => ({
      push: jest.fn(),
    }),
  };
});

describe('<SignInScreen />', () => {
  //   it('renders', () => {
  //     const { container } = render(<SignInScreen />);

  //     expect(container.firstChild).toMatchSnapshot();
  //   });
  //   it('changes input values on change', () => {
  //     const { container, getByPlaceholderText } = render(<SignInScreen />);

  //     const email = 'test@test.com';
  //     const password = '123';

  //     userEvent.type(getByPlaceholderText('Email'), email);
  //     userEvent.type(getByPlaceholderText('Password'), password);

  //     expect(getByPlaceholderText('Email')).toHaveValue(email);
  //     expect(getByPlaceholderText('Password')).toHaveValue(password);

  //     expect(container.firstChild).toMatchSnapshot();
  //   });
  it('sign ins user', async () => {
    const { container, getByTestId } = render(<SignInScreen />);
    const form = getByTestId('form');

    fireEvent.submit(form);

    expect(container.firstChild).toMatchSnapshot();
  });
});
