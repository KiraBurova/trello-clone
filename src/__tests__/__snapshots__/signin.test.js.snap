// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignInScreen /> sign ins user 1`] = `
<div
  class="signin__Container-sc-1tsapn2-0 cmxyBq"
>
  <form
    class="form__Base-sc-1grfi8t-0 hLGYyj"
    data-testid="form"
  >
    <h2
      class="form__Title-sc-1grfi8t-1 ekfsMg"
    >
      Sign In
    </h2>
    <input
      class="form__Input-sc-1grfi8t-2 czvKJK"
      name="email"
      placeholder="Email"
    />
    <input
      class="form__Input-sc-1grfi8t-2 czvKJK"
      name="password"
      placeholder="Password"
      type="password"
    />
    <button
      class="form__Button-sc-1grfi8t-3 ijFtjC"
      disabled=""
    >
      Sign In
    </button>
    <p
      class="form__ErrorText-sc-1grfi8t-4 ioXfqK"
    />
    <p
      class="form__LoadingText-sc-1grfi8t-5 hVQvTy"
    />
  </form>
</div>
`;
